portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,18 +20,51 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
-import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
 
-/** Servlet that returns list of greeting messages. TODO: modify this file to handle comments data */
+/** Servlet that handles comment data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private static final ImmutableList<String> HELLO_MESSAGES = ImmutableList.of("Hello Megan!", "Hello there!", "Hello!");
+  private ArrayList<Comment> comments = new ArrayList<Comment>();
+
+   @Override
+   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+     response.setContentType("application/json;");
+     Gson gson = new Gson();
+     String json = gson.toJson(comments);
+     response.getWriter().println(json);
+   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json;");
-    Gson gson = new Gson();
-    String json = gson.toJson(HELLO_MESSAGES);
-    response.getWriter().println(json);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = getParameter(request, "comment", "");
+    String displayName = getParameter(request, "name", "");
```

Comment:
nit: I generally prefer to define all literals as constants at the top of the page, in case we need it again. Even if we only ever use them once, it's still nice to separate logic from constants (in case the "comment" key ever changes elsewhere, we can easily find it and change it without sifting through the code).